### DISCUSSION TITLE
[SOLVED] Cannot update? Remaining backlog?

### DISCUSSION THREAD

**Discussion by ArvyRogerio** [osm-search/Nominatim#3455] [Q&A] [answered]
**Describe the bug**
After install, running "replication --once" returns that is already updated. 
Returns a "Remaining backlog" message, with progressive time.

**Software Environment (please complete the following information):**
- Nominatim version: 4.4.0
- Postgresql version: 15
- Postgis version: 3.3
- OS: Ubuntu 22.04

**Hardware Configuration (please complete the following information):**
- RAM: 32 Gb
- number of CPUs: 8
- type and size of disks: 2 Tb SATA

**Nominatim Configuration:**

Update script:

> #!/bin/bash
> 
> export NOMINATIM_REPLICATION_URL="http://download.geofabrik.de/south-america/brazil-updates"
> export NOMINATIM_REPLICATION_UPDATE_INTERVAL=86400
> export NOMINATIM_REPLICATION_RECHECK_INTERVAL=3600
> 
> cd /dados/nominatim/nominatim-planet
> nominatim replication --once

Returns:

> 2024-06-24 09:49:40: Using project directory: /dados/nominatim/nominatim-planet
> 2024-06-24 09:49:44: Update completed. Import: 0:00:00. Total: 0:00:00. Remaining backlog: 7 days, 19:30:12.

Running "--init" (as suggested in another topic), no difference.

**Comment by mtmail** [comment]
Can you post your `/dados/nominatim/nominatim-planet/.env` file? I wonder if it contains lines which overwrite the environment variables.

What does this database query print? `SELECT * FROM import_status;`

**Comment by mtmail** [comment]
Also what does `SELECT * FROM import_osmosis_log;` print?

**Comment by ArvyRogerio** [comment]
I have no .env file in there, I set the 3 "export" in the bash script. Something missing then?

EDIT: I've copied /usr/local/etc/nominatim/env.defaults to .env, no difference too.

**Comment by ArvyRogerio** [comment]
> nominatim=# SELECT * FROM import_osmosis_log;
>  batchend | batchseq | batchsize | starttime | endtime | event
> ----------+----------+-----------+-----------+---------+-------
> (0 rows)

> nominatim=# SELECT * FROM import_status;
>      lastimportdate     | sequence_id | indexed
> ------------------------+-------------+---------
>  2024-06-16 14:19:32-03 |     6136057 | t
> (1 row)

**Comment by lonvia** [comment, marked as answer]
You still have done something wrong when running 'nominatim replication --init'. It has not picked up the NOMINATIM_REPLICATION_URL correctly. The setup is still for the global replication source.

Please use an `.env` file in your project directory instead of the export statements. This avoids such confusion and errors.

**Reply by ArvyRogerio** [reply]
Ok, I need to drop the db and redo? Or just create the .env file, run --init again?

**Reply by lonvia** [reply]
Just run --init again. Afterwards, --once should work fine.

**Reply by ArvyRogerio** [reply]
Working! 😎

Thanks a lot for helping me.
